### PR TITLE
Fix config-flow migration and add new biomass boiler properties

### PR DESCRIPTION
**Pull request by lein1013**
Fix for https://github.com/LavermanJJ/home-assistant-solarfocus/issues/46

### CODE REVIEW

File: custom_components/solarfocus/sensor.py [hunk 1]
```diff
@@ -318,6 +318,7 @@ def native_value(self):
     SensorEntityDescription(
         key="single_charge",
         icon="mdi:pump",
+        device_class="solarfocus__bocharge"
```
Comment (LavermanJJ) [inline custom_components/solarfocus/sensor.py:321]: to be removed. see below

File: custom_components/solarfocus/sensor.py [hunk 2]
```diff
@@ -519,6 +520,7 @@ def native_value(self):
     SensorEntityDescription(
         key="message_number",
         icon="mdi:message-text-outline",
+        device_class="solarfocus_pbmessagenumber"
```
Comment (LavermanJJ) [inline custom_components/solarfocus/sensor.py:523]: The way translations work changed in 2023.4. A translation key will be automatically generated 
https://github.com/LavermanJJ/home-assistant-solarfocus/blob/6140137d6f7c6da526efae0d66493185594d042c/custom_components/solarfocus/entity.py#L58-L67

-> bo_message_number would be the translation key to use then in the `strings.json` file 
 e.g. 
https://github.com/LavermanJJ/home-assistant-solarfocus/blob/6140137d6f7c6da526efae0d66493185594d042c/custom_components/solarfocus/strings.json#L63

And also in translations/en.json and translations/de.json

File: custom_components/solarfocus/sensor.py [hunk 3]
```diff
@@ -564,6 +566,27 @@ def native_value(self):
...
+        key="pellet_usage_total",
+        native_unit_of_measurement=MASS_KILOGRAMS,
+        icon="mdi:alpha-t-box",
+        device_class=SensorDeviceClass.WEIGHT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="heat_energy_total",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        icon="mdi:fire-circle",
```
Comment (LavermanJJ) [inline custom_components/solarfocus/sensor.py:586]: to be consistent I'd use "mdi:meter-gas" for thermal energy cf

https://github.com/LavermanJJ/home-assistant-solarfocus/blob/6140137d6f7c6da526efae0d66493185594d042c/custom_components/solarfocus/sensor.py#L371-L377

File: hacs.json [hunk 4]
```diff
@@ -1,9 +1,9 @@
 {
-    "name": "Solarfocus eco manager-touch",
+    "name": "Solarfocus eco manager-touch for eco top light",
```
Comment (LavermanJJ) [inline hacs.json:2]: Please remove. see above

**Review by LavermanJJ — CHANGES_REQUESTED** [review]
Thanks for providing the PR 👍  I've provided some comments.

**Comment by LavermanJJ** [comment]
Also fixes #42

File: README.md [hunk 5]
```diff
@@ -9,7 +9,8 @@ Custom component for Home-Assistant to integrate [Solarfocus](https://www.solarf
 [![hacs_badge](https://img.shields.io/badge/HACS-Default-41BDF5.svg?style=for-the-badge)](https://github.com/hacs/integration)
 
 - Install [Home Assistant Community Store (HACS)](https://hacs.xyz/)
-- Add custom repository https://github.com/lavermanjj/home-assistant-solarfocus to HACS
+- Add custom repository https://github.com/lavermanjj/home-assistant-solarfocus to HACS (original)
+- or forked version for eco top light https://github.com/lein1013/home-assistant-solarfocus
```
[the remark this answers is not in the document]
Comment (lein1013) [reply README.md:13]: Ups, that came by accident - I just wanted to do the pull request for init - I've just tested the enums of https://github.com/LavermanJJ/home-assistant-solarfocus/issues/42 and will do this there. I'll try to clean up

[the remark this answers is not in the document]
Comment (LavermanJJ) [reply README.md:13]: Even better - Let's split it into two PRs, one for the init, one for #42 👍

**Comment by lein1013** [comment]
Since this PR got a bit too large, I propose **not** to merge and just close it
I need to figure out how to make a clean sync of the fork again and then just add `CONF_FRESH_WATER_MODULE` in the import

```
from .const import (
    CONF_PHOTOVOLTAIC,
    CONF_HEATPUMP,
    CONF_PELLETSBOILER,
    CONF_BOILER,
    CONF_BUFFER,
    CONF_HEATING_CIRCUIT,
    CONF_SOLAR,
    CONF_SOLARFOCUS_SYSTEM,
    CONF_FRESH_WATER_MODULE,
    DATA_COORDINATOR,
    DOMAIN,
)
```

**Pull request closed without merging**